cumulus/polkadot-parachain/src/service.rs

```diff
@@ -518,23 +519,24 @@ impl<R> BuildOnAccess<R> {
 
 struct Verifier<Client, AuraId> {
 	client: Arc<Client>,
-	aura_verifier: BuildOnAccess<Box<dyn VerifierT<Block>>>,
+	aura_verifier: Mutex<BuildOnAccess<Box<dyn VerifierT<Block>>>>,
```

Comment:
Hmm I think we can remove this altogether. We could make this thing simpler by just removing `Mutex` and `BuildOnAccess` and directly instantiating the aura verifier. All chains that use it will sooner or later transition to aura anyway, so I don't think we save much by instantiating on demand. The call to `verify` itself is anyway guarded by a check of aura API.

Reply:
Makes sense. Also contruction of verifier seems really cheap, so this is just over-engineering. Thanks for suggestion!